doc/user_guide/statements/select.md

```diff
@@ -165,3 +167,29 @@ select.from().table("t");
 select.orderBy(column("t", "city"), column("t", "price"))
         .nullsFirst().asc();
 ```
+
+## Creating Functions
+
+When you need to use special functions like analytic functions, you can add them to a statement like this:
+
+```java
+Function function = ... // create function
+Select select = factory.select().function(function, "<column>");
+```
+
+### Analytic functions
+
+#### Keywords `DISTINCT` and `ALL`
+
+You create an analytic function with a keyword `DISTINCT` or `ALL` like this:
+
+```java
+Function function = AnalyticFunction.of(ExasolAnalyticFunction.ANY, Keyword.DISTINCT,
```

Comment:
Maybe we could do something like a builder here too? It's lots of parameters and I did not get immediately what the mean.